easybuild/easyconfigs/p/PyTorch/PyTorch-1.7.0-fosscuda-2019b-Python-3.7.4-imkl.eb

```diff
@@ -0,0 +1,325 @@
+name = 'PyTorch'
+version = '1.7.0'
+local_pyversionsuffix = '-Python-%(pyver)s'
+versionsuffix = '-Python-%(pyver)s-imkl'
+
+homepage = 'https://pytorch.org/'
+description = """Tensors and Dynamic neural networks in Python with strong GPU acceleration.
+PyTorch is a deep learning framework that puts Python first."""
+
+toolchain = {'name': 'fosscuda', 'version': '2019b'}
+
+osdependencies = [OS_PKG_IBVERBS_DEV]
+
+builddependencies = [
+    ('CMake', '3.15.3'),
+    ('hypothesis', '4.44.2', local_pyversionsuffix),
+    ('libarchive', '3.4.2'),
```

Comment:
This shouldn't be required. The error message from https://github.com/easybuilders/easybuild-easyconfigs/pull/11636#issuecomment-735030882 hints at cmake not pytorch so if at all it should be added to the cmake EC but first it should be checked why it uses the system cmake

Reply:
Just to check, is it `libarchive` or `cmake` that shouldn't be required in `builddependencies`?

Reply:
libarchive. The CMake 3.15 should need it. I see CMake-3.18.4-GCCcore-10.2.0.eb added it so I guess your system CMake is newer than 3.15 and something is wrong so the system cmake gets picked up

Reply:
Hm, ick. No idea why this happens. My familiarity with CMake isn't great. :-(

Reply:
Nothing with CMake here, only with the CMake module. As you posted the PyTorch build somehow uses /usr/bin/cmake3 which is clearly wrong. So try to load the CMake/3.15.3 module and run `which cmake` and `which cmake3` to check if the module is broken.

Reply:
Aha! Good call!

```
terjekv@ninhursaga:~$ module load CMake/3.15.3-GCCcore-8.3.0
terjekv@ninhursaga:~$ type -a cmake
cmake is /opt/uio/modules/rhel8/easybuild/software/CMake/3.15.3-GCCcore-8.3.0/bin/cmake
cmake is /usr/bin/cmake
terjekv@ninhursaga:~$ type -a cmake3
cmake3 is /usr/bin/cmake3
```

Reply:
And I checked the PyTorch code: `CMAKE = find_executable('cmake3') or find_executable('cmake')`
Looks like we need an update to the CMake easyblock

Reply:
Yeah, good catch. I doubt I have time to sort it right now. Make an issue of it and first one gets a cookie?

Reply:
My cookie please: https://github.com/easybuilders/easybuild-easyblocks/pull/2259

Reply:
🍪 !

Reply:
Updated in https://github.com/easybuilders/easybuild-easyconfigs/pull/11786/commits/117a9bd3d9d4bc08206cd3af80eb5d226892d073 to not need libarchive, which I didn't include in the PR anyway... I was clearly ahead of myself this morning.